gulpfile.js

```diff
@@ -55,8 +55,14 @@ gulp.task('deploy', ['copy'], () => {
 
   options.ptr = opts.ptr || false
   options.branch = opts.branch || 'default'
-  options.email = opts.email || opts.username
-  options.password = opts.password
+
+  if (opts.token) {
```

Comment:
Just a note, gulp-screeps already does this check internally, so you can set email, password, and token and it will use the token first if it exists.